File: Hello Algorithm/Hello Algorithm_Java/src/com/algorithm/stack/Stack.java
package com.algorithm.stack;

import com.algorithm.List.ArrayList;

public class Stack <E>{
    //内部使用list(arrayList 和 linkedList均可)
    private ArrayList<E> list;
    public Stack() {
        list = new ArrayList<>();
    }

    /**
     * 获取栈内节点的数量
     * @return
     */
    public int size() {
        return list.size();
    }

    /**
     * 获取栈是否是空的
     * @return
     */
    public boolean isEmpty() {
        return list.isEmpty();
    }

    /**
     * 入栈
     * @param element
     */
    public void push(E element) {
        list.add(element);
    }

    /**
     * 出栈
     * @return
     */
    public E pop() {
        return list.remove(list.size() - 1);
    }


    /**
     * 获取栈顶元素
     * @return
     */
    public E top() {
        return list.get(list.size() - 1);
    }

    /**
     * 清空栈
     */
    public void clear() {
        list.clear();
    }

    @Override
    public String toString() {
        return list.toString();
    }
}

File: Hello Algorithm/Hello Algorithm_Swift/Hello Algorithm/Hello Algorithm/栈/Stack.swift
//
//  Stack.swift
//  Hello Algorithm
//
//  Created by mac on 2021/9/9.
//



class Stack <E : Equatable> : CustomStringConvertible {
    private var _list : LinkedList<E>
    init() {
        _list = LinkedList()
    }
    
    
    /// 获取栈内节点的数量
    /// - Returns:
    public func size() -> Int {
        _list.getSize()
    }
    
    
    /// 判断栈是否为空
    /// - Returns:
    public func isEmpty() -> Bool {
        _list.isEmpty()
    }
    
    /// 入栈
    /// - Parameter element:
    /// - Returns:
    public func push(element: E) -> Void {
        _list.addElement(element: element)
    }
    
    /// 出栈
    /// - Returns:
    public func pop() -> E {
        _list.remove(at: (_list.getSize() - 1))
    }
    
    /// 获取栈顶元素
    /// - Returns:
    public func top() -> E {
        _list.getElement(at: (_list.getSize() - 1))
    }
    
    /// 清空栈
    public func clear() {
        _list.clear()
    }
    
    var description: String {
        _list.description
    }
    
    
}
